### PR TITLE
D3ASIM-1873: Update one sided market rate threshold

### PR DESCRIPTION
**Pull request by Hassan754**

### CODE REVIEW

File: integration_tests/steps/infinite_bus_tests.py
Comment (spyrostz) [inline integration_tests/steps/infinite_bus_tests.py:38]: Reminder to remove before merging

**Review by spyrostz — APPROVED** [review]
LGTM, would be great to write an outcome of the discussion on the Jira ticket so that the research team understands what is implemented here.

**Review by hannesdiedrich — APPROVED** [review]
LGTM once Spyros comments are addressed.

File: src/d3a/setup/strategy_tests/load_const_price_increase.py
Comment (hannesdiedrich) [inline src/d3a/setup/strategy_tests/load_const_price_increase.py:34]: Please remove this and also from the file were you copy pasted this from. thanks!

File: integration_tests/load_tests.feature
Comment (hannesdiedrich) [inline integration_tests/load_tests.feature:37]:
```suggestion
  Scenario: Load strategy increases its accepted offers price in one sided market
```

File: integration_tests/steps/load_tests.py
Comment (hannesdiedrich) [inline integration_tests/steps/load_tests.py:72]: This does not check whether the offer price really got increased. There is only one trade in each market and probably they have all the same price. 
I think you can only check it with a unittest. Please correct me, if I overlook something. Thanks!

**Review by hannesdiedrich — COMMENTED** [review]
Added some comments, LGTM after addressed

File: src/d3a/models/strategy/storage.py
Comment (spyrostz) [inline src/d3a/models/strategy/storage.py:516]: Does not hurt to change that, however I am curious why this was needed in the first place. Essentially, what this did was a sanity check in order to always make sure that the bid rate will never surpass the final rate (in order to be able to cover potential problems with the rate calculation like e.g. border conditions due to floating point precision) , therefore I think it would be better if you left it at peace unless there is a specific reason to change it.

**Review by spyrostz — APPROVED** [review]
1 comment, LGTM once this and Hannes' comments are fixed.

File: src/d3a/models/strategy/storage.py
Comment (faizan2590) [reply src/d3a/models/strategy/storage.py:516]: sorry to jump in, Spyros AFAI, the sanity check is already part of [bid_update.get_updated_rate](https://github.com/gridsingularity/d3a/blob/master/src/d3a/models/strategy/update_frequency.py#L135) implementation

Comment (spyrostz) [reply src/d3a/models/strategy/storage.py:516]: Then it is all good here, leave it as it is

**Review by spyrostz — APPROVED** [review]
LGTM

File: tests/test_strategy_load_hours.py
Comment (hannesdiedrich) [inline tests/test_strategy_load_hours.py:520]: Hmm, sorry to be relentless here. But this only checks for the functionality of `load.bid_update.get_updated_rate`, but not the correct bid placement. 
What happened to my suggestion, to access the `market.bids` member in the integrationtests?

Comment (hannesdiedrich) [reply tests/test_strategy_load_hours.py:520]: Ah, yes, you are right. And now I understand your unittest. Agreed.

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

**Pull request merged**
